compiler/src/openxla/compiler/nvgpu/Conversion/ConvertHLOToCUDNN.cpp

```diff
@@ -58,107 +59,51 @@ static FailureOr<Layout> getCudnnTensorLayout(int64_t batchDim,
+    rewriter.create<IREE::Util::GlobalStoreOp>(loc, handle, *globalHandle);
+    rewriter.create<IREE::Util::InitializerReturnOp>(loc);
+
+    return *globalHandle;
   }
 
-  // Create the cudnn.graph op with an empty region.
-  SymbolTable symbolTable(func->getParentOp());
-  rewriter.setInsertionPoint(func);
-  GraphOp graph = rewriter.create<GraphOp>(root->getLoc(), funcType,
-                                           /*arg_attrs=*/ArrayAttr{},
-                                           /*res_attrs=*/ArrayAttr{});
-  graph.setName(root->getName().getStringRef());
-  symbolTable.insert(graph);
-
-  // Clone ops into region.
-  rewriter.setInsertionPointToStart(graph.addEntryBlock());
-  IRMapping mapping;
-  for (size_t index = 0; index < arguments.size(); ++index) {
-    Value arg = arguments[index];
-    auto castOp = rewriter.create<UnrealizedConversionCastOp>(
-        arg.getLoc(), arg.getType(), graph.getArgument(index));
-    mapping.map(arg, castOp.getResult(0));
-  }
-  ValueRange results;
-  for (Operation* op : llvm::reverse(ops)) {
-    results = rewriter.clone(*op, mapping)->getResults();
-    mapping.map(op->getResults(), results);
-  }
-  auto castOp = rewriter.create<UnrealizedConversionCastOp>(
-      result.getLoc(), funcType.getResults(), results);
-  rewriter.create<ReturnOp>(root->getLoc(), castOp.getResults());
-
-  // Create global handle before any other initialization.
-  rewriter.setInsertionPointToStart(m.getBody());
-  Location loc = root->getLoc();
-  std::string globalHandleName = (graph.getName() + ".handle").str();
-  auto handleTy = rewriter.getType<HandleType>();
-  auto globalHandle = rewriter.create<IREE::Util::GlobalOp>(
-      loc, globalHandleName, /*isMutable=*/false, handleTy);
-
-  // Create global handle initializer.
-  auto initializer = rewriter.create<IREE::Util::InitializerOp>(loc);
-  rewriter.setInsertionPointToStart(initializer.addEntryBlock());
-  auto device = rewriter.create<IREE::HAL::ExSharedDeviceOp>(loc);
-  auto handle = rewriter.create<HandleOp>(loc, device);
-  rewriter.create<IREE::Util::GlobalStoreOp>(loc, handle, globalHandle);
-  rewriter.create<IREE::Util::InitializerReturnOp>(loc);
-
-  // Replace root with cudnn.call op.
-  rewriter.setInsertionPoint(root);
-  auto loadedHandle = rewriter.create<IREE::Util::GlobalLoadOp>(
-      loc, handleTy, globalHandle.getSymNameAttr());
-  rewriter.replaceOpWithNewOp<CallOp>(
-      root, result.getType(), graph.getName(), loadedHandle,
-      ArrayRef<Value>(arguments.begin(), arguments.size()));
-  return success();
-}
+ private:
+  std::optional<IREE::Util::GlobalOp> globalHandle;
+};
+
+/// Outline HLO ops into CUDNN graphs.
 
 // Returns the 'min' value of clamp 'op', if it can be converted to relu.
-static FailureOr<llvm::APFloat> matchRelu(stablehlo::ClampOp op,
-                                          PatternRewriter& rewriter) {
+FailureOr<llvm::APFloat> matchRelu(stablehlo::ClampOp op,
```

Comment:
IREE/OpenXLA follows LLVM style under compiler directory, and it suggests not to wrap everything into anonymous namespace: https://llvm.org/docs/CodingStandards.html#anonymous-namespaces and use static functions instead.

Reply:
Done.